docs/modules/candid-guide/index.adoc

```diff
@@ -0,0 +1,1098 @@
+  circle : record { radius : float64 };
+  rectangle : record { width : float64; height : float64 };
+  "💬" : text;
+};
+....
+is either a dot, or a circle (with a radius), or a rectangle (with dimensions) or a speech bubble (with a text, and showing off unicode label names).
+
+The lables in variants are, just like the labels in records, actually numbers, and string labels refer to their hash value.
+
+Often, some or all of the the tags do not carry data. It is idiomatic to then use the `null` type, as in the `dot` above. In fact, Candid encourages this by allowing you to omit the `: null` type annotation in variants, so
+....
+type season = variant { spring; summer; fall; winter }
+....
+is equivalent to
+....
+type season = variant { spring : null; summer: null; fall: null; winter : null }
+....
+and used to represent enumerations.
+
+The type `variant {}` is legal, but has no values. If that is the intention, the <<type-empty,`empty` type>> may be more appropriate.
+
+Type syntax::
++
+....
+variant {}
+variant { ok : nat; error : text }
+variant { "name with spaces" : nat; "unicode, too: ☃" : bool }
+....
+
+Textual syntax::
++
+....
+variant { ok = 42 }
+variant { "unicode, too: ☃" = true }
+....
+
+Subtypes::
+
+Subtypes of a variant type are variant types with some tags removed, and the type of some tags themselves changed to a subtype.
++
+If you want to be able to _add_ new tags in variants in a method result, you can do so if the variant it itself wrapped in `opt …`. This requires planning ahead! When you design an interface, instead of writing
++
+....
+service { get_member_status (member_id : nat) -> (variant {active; expired}); }
+....
++
+better use
++
+....
+service { get_member_status (member_id : nat) -> (opt variant {active; expired}); }
+....
++
+This way, if you later need to add a `honory` membership status, you can expand the list of statuses. Old clients will receive unknown fields as `null`.
+
+Supertypes::
+
+Supertypes of a variant types are variants with additional tags, and maybe the type of some tags changed to a supertype.
+
+
+Corresponding Motoko type::
+
+Variant types are represented as Motoko variant types, e.g.
++
+....
+type Shape = {
+  #dot : ();
+  #circle : { radius : Float };
+  #rectangle : { width : Float; height : Float };
+  #_2669435721_ : Text;
+};
+....
++
+Note that if the type of a tag is `null`, this corresponds to `()` in Motoko, to preserve the mapping between the respecitve idiomatic ways to model enumerations as variants.
+
+Corresponding Rust type::
+
+TODO
+
+Corresponding JavaScript values::
+
+TODO
+
+
+[#type-func]
+=== Type func (…) -> (…)
+
+Candid is designed to support higher-order use cases, where a service may receive or provide references to other serivces and their methods. The `func` type lies at the code or this: It indicates the function’s _signature_ (argument and results types, annotations), and values of this type are references to functions with that signature.
```

Comment:
```suggestion
Candid is designed to support higher-order use cases, where a service may receive or provide references to other services and their methods. The `func` type lies at the code or this: It indicates the function’s _signature_ (argument and results types, annotations), and values of this type are references to functions with that signature.
```

Reply:
```suggestion
Candid is designed to support higher-order use cases, where a service may receive or provide references to other services and their methods. The `func` type lies at the heart of this: It indicates the function’s _signature_ (argument and results types, annotations), and values of this type are references to functions with that signature.
```

Reply:
I meant “core”, but Andreas already fixed that one.